tests/dialects/test_dlti.py

```diff
@@ -0,0 +1,270 @@
+import pytest
+
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    Float32Type,
+    FloatAttr,
+    IntegerAttr,
+    StringAttr,
+    i32,
+)
+from xdsl.dialects.dlti import (
+    DataLayoutEntryAttr,
+    DataLayoutSpecAttr,
+    DLTIEntryMap,
+    MapAttr,
+    TargetDeviceSpecAttr,
+    TargetSystemSpecAttr,
+)
+from xdsl.ir import Attribute, TypeAttribute
+from xdsl.utils.exceptions import VerifyException
+
+
+def test_data_layout_entry():
+    # Test passing strings for key and value
+    entry = DataLayoutEntryAttr("test", "V")
+    assert isinstance(entry.key, StringAttr)
+    assert isinstance(entry.value, StringAttr)
+    assert entry.key.data == "test"
+    assert entry.value.data == "V"
+
+    # Test passing string for key and int for value
+    entry = DataLayoutEntryAttr("test2", 12)
+    assert isinstance(entry.key, StringAttr)
+    assert entry.key.data == "test2"
+    assert isinstance(entry.value, IntegerAttr)
+    assert entry.value.value.data == 12  # pyright: ignore
+
+    # Test passing string for key and float for value
+    entry = DataLayoutEntryAttr("test3", 99.45)
+    assert isinstance(entry.key, StringAttr)
+    assert isinstance(entry.value, FloatAttr)
+    assert entry.key.data == "test3"
+    assert round(entry.value.value.data, 2) == 99.45  # pyright: ignore
+
+    # Test passing type for key and string for value
+    entry = DataLayoutEntryAttr(i32, "test")
+    assert isinstance(entry.key, TypeAttribute)
+    assert isinstance(entry.value, StringAttr)
+    assert entry.key == i32
+    assert entry.value.data == "test"
+
+    # Test passing StringAttr for key and value
+    entry = DataLayoutEntryAttr(StringAttr("k"), StringAttr("v"))
+    assert isinstance(entry.key, StringAttr)
+    assert isinstance(entry.value, StringAttr)
+    assert entry.key.data == "k"
+    assert entry.value.data == "v"
+
+
+def test_incorrect_data_layout_entry():
+    with pytest.raises(VerifyException):
+        entry = DataLayoutEntryAttr(12, "V")  # pyright: ignore
```

Comment:
we tend not to add tests like this since the assumption is that the user is using a type checker that will prevent them from calling the function with the wrong type anyway, the overall approach is that programs with pyright errors have undefined behaviour

Reply:
Thanks, I've broken these out into separate tests